src/components/audio/AudioContextProvider.js

```diff
@@ -0,0 +1,369 @@
+  const [playlist, setPlaylist] = React.useState(() =>
+    buildPlaylistFromCollectibles(collectibles)
+  );
+
+  // Deeply-memoized playlist management. (Avoid reallocation of playlists
+  // when the content is effectively identical.)
+  React.useEffect(() => {
+    const maybeNextPlaylist = buildPlaylistFromCollectibles(collectibles);
+    if (!isEqual(playlist, maybeNextPlaylist)) {
+      setPlaylist(maybeNextPlaylist);
+    }
+  }, [buildPlaylistFromCollectibles, collectibles, setPlaylist, playlist]);
+
+  const [currentSound, setCurrentSound] = React.useState(null);
+
+  const dispatch = useDispatch();
+  const { currentlyPlayingAsset, autoplay } = useSelector(
+    ({ audio: { currentlyPlayingAsset, autoplay } }) => ({
+      autoplay,
+      currentlyPlayingAsset,
+    }),
+    isEqual
+  );
+
+  const pickRandomAsset = React.useCallback(() => {
+    const choices = playlist.filter(e => e !== currentlyPlayingAsset);
+    const { length: numberToChoose } = choices;
+    return numberToChoose > 0
+      ? choices[Math.floor(Math.random() * choices.length)]
+      : playlist[0];
+  }, [playlist, currentlyPlayingAsset]);
+  const pickNextAsset = React.useCallback(() => {
+    const nextIndex =
+      (playlist.indexOf(currentlyPlayingAsset) + 1) % playlist.length || 0;
+    return playlist[nextIndex];
+  }, [playlist, currentlyPlayingAsset]);
+
+  const playAsset = React.useCallback(
+    asset => {
+      setCurrentSound(maybeCurrentSound => {
+        !!maybeCurrentSound && maybeCurrentSound.setVolume(0);
+        return maybeCurrentSound;
+      });
+      return dispatch(setCurrentPlayingAsset(asset));
+    },
+    [dispatch, setCurrentSound]
+  );
+
+  const stopPlayingAsset = React.useCallback(
+    () => dispatch(setCurrentPlayingAsset(null)),
+    [dispatch]
+  );
+
+  const isPlayingAsset = !!currentlyPlayingAsset;
+
+  const [isPlayingAssetPaused, setIsPlayingAssetPaused] = React.useState(false);
+
+  React.useEffect(() => {
+    const i = setInterval(() => {
+      const nextIsPlayingAssetPaused =
+        !!currentSound && currentlyPlayingAsset && !currentSound._playing;
+      if (nextIsPlayingAssetPaused !== isPlayingAssetPaused) {
+        setIsPlayingAssetPaused(nextIsPlayingAssetPaused);
+      }
+    }, 10);
+    return () => clearInterval(i);
+  }, [
+    currentSound,
+    setIsPlayingAssetPaused,
+    currentlyPlayingAsset,
+    isPlayingAssetPaused,
+  ]);
+
+  const shouldLoadSoundByUri = React.useCallback(async uri => {
+    const sound = await new Promise((resolve, reject) => {
+      const loadedSound = new Sound(uri, null, err => {
+        if (err) {
+          return reject(err);
+        }
+        return resolve(loadedSound);
+      });
+    });
+    return sound;
+  }, []);
+
+  const fadeTo = React.useCallback(async (nextSound, toValue = 0) => {
+    if (!nextSound) {
+      return;
+    } else if (typeof nextSound.getVolume !== 'function') {
+      logger.error(
```

Comment:
do we need this error handling? How can one get confused when this name is quite obsious. 
I don't have a strong opinion tho